netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -258,6 +261,45 @@ _ebpf_sock_addr_get_current_logon_id(_In_ const bpf_sock_addr_t* ctx)
     return logon_id;
 }
 
+static int
+_ebpf_sock_addr_set_redirect_context(_In_ const bpf_sock_addr_t* ctx, _In_ void* data, _In_ uint32_t data_size)
+{
+    int return_value = 0;
+    net_ebpf_sock_addr_t* sock_addr_ctx = CONTAINING_RECORD(ctx, net_ebpf_sock_addr_t, base);
+    void* redirect_context = NULL;
+
+    // This function is only supported at the connect redirect layer.
+    if (sock_addr_ctx->hook_id != EBPF_HOOK_ALE_CONNECT_REDIRECT_V4 &&
+        sock_addr_ctx->hook_id != EBPF_HOOK_ALE_CONNECT_REDIRECT_V6) {
+        return_value = -1;
+        goto Exit;
+    }
+
+    // Allocate buffer to store redirect context.
+    redirect_context = ExAllocatePoolUninitialized(NonPagedPoolNx, data_size, NET_EBPF_EXTENSION_POOL_TAG);
```

Comment:
If `data_size` is 0, we will do 0-size allocation. I think we should fail the call if size is 0.